src/pages/wallet-overview/wallet-overview.spec.ts

```diff
@@ -514,5 +456,493 @@ describe("WalletOverviewPage", () => {
+
+    comp.writeFileMobile("location", "filename", "keystoredata", {replace: true}, "android").then(data => {
+      expect(toastController.create).toHaveBeenCalledWith({
+        message: jasmine.any(String),
+        position: "bottom",
+        showCloseButton: true,
+        closeButtonText: "Ok"
+      });
+      expect(toast.present).toHaveBeenCalled();
+      done();
+    });
+  });
+
+  it("should show that there are no transactions if it returns an empty array", (done) => {
+    spyOn(transactionHistoryService, "getTransactionHistory").and.returnValue(Promise.resolve([]));
+
+    comp.getTransactionHistory("").then(data => {
+      expect(comp.noTransactionHistoryVisibility).toBe("shown");
+      expect(comp.transactionHistoryVisibility).toBe("hidden");
+      done();
+    });
+  });
+
+  it("should show an alert and retry button when the transaction history could not be retrieved", (done) => {
+    let alert = new MockAlert();
+
+    spyOn(transactionHistoryService, "getTransactionHistory").and.returnValue(Promise.reject([]));
+    spyOn(alertController, "create").and.returnValue(alert);
+    spyOn(alert, "present");
+
+    comp.getTransactionHistory("").then(data => {
+      expect(alertController.create).toHaveBeenCalled();
+      expect(alert.present).toHaveBeenCalled();
+      done();
+    });
+  });
+
+  it("should show a retry modal for the exchanges when the request returns an empty OK response", (done) => {
+    spyOn(exchangeService, "getAvailableExchanges").and.returnValue(Promise.resolve([]));
+    spyOn(comp, "getAvailableExchanges").and.callThrough();
+    spyOn(comp, "presentExchangesRetryButton");
+
+    comp.getAvailableExchanges().then(data => {
+      expect(comp.presentExchangesRetryButton).toHaveBeenCalled();
+      done();
+    });
+  });
+
+  it("should push zero balances for XSM and XSP when the JSON is null or has no keys at all", (done) => {
+    comp.currentWallet = getDummyWallet();
+
+    spyOn(walletBalancesService, "getWalletBalance").and.returnValue(Promise.resolve(null));
+
+    comp.getWalletBalance("").then(data => {
+      expect(comp.balances[0]).toEqual({currency: "XSM", amount: 0, valueAmount: 0});
+      expect(comp.balances[1]).toEqual({currency: "XSP", amount: 0, valueAmount: 0});
+      done();
+    });
+  });
+
+  it("should show a retry modal when retrieving the balances has failed", (done) => {
+    let alert = new MockAlert();
+
+    spyOn(walletBalancesService, "getWalletBalance").and.returnValue(Promise.reject(""));
+    spyOn(alertController, "create").and.returnValue(alert);
+    spyOn(alert, "present");
+
+    comp.getWalletBalance("").then(data => {
+      expect(alertController.create).toHaveBeenCalled();
+      expect(alert.present).toHaveBeenCalled();
+      done();
+    });
+  });
+
+  it("should set the currencies correct for the picked exchange and picked currency and when the currency doesn't exist on the exchange it should pick the first one on the exchange", () => {
```

Comment:
Perhaps it is best to split this test into two?

Test 1: should set the currencies correct for the picked exchange and picked currency
Test 2: when the currency doesn't exist on the exchange it should pick the first one on the exchange